languages/rust/generic/Parse_rust_tree_sitter.ml

```diff
@@ -674,10 +657,23 @@ and map_range_pattern_bound (env : env) (x : CST.anon_choice_lit_pat_0884ef0) :
       G.PatConstructor (name, [])
 
 and map_meta_argument (env : env) (x : CST.anon_choice_meta_item_fefa160) :
-    rust_meta_argument =
+    G.argument =
   match x with
-  | `Meta_item x -> MetaArgMetaItem (map_meta_item env x)
-  | `Lit x -> MetaArgLiteral (map_literal env x)
+  (* TODO: With modifications to the rust grammar this could be a lot better. *)
+  | `Meta_item x -> map_meta_item_to_argument env x
+  | `Lit x -> G.(Arg (L (map_literal env x) |> e))
+
+and map_meta_item_to_argument (env : env) ((v1, v2) : CST.meta_item) :
+    G.argument =
+  let name = map_path_name env v1 in
+  match v2 with
+  | None -> G.(Arg (N name |> e))
+  | Some (`Meta_args x) ->
+      G.(Arg (Call (N name |> e, map_meta_arguments env x) |> e))
+  | Some (`EQ_lit (v1, v2)) ->
+      let _equals = token env v1 (* "=" *) in
+      let lit = map_literal env v2 in
+      G.(OtherArg (fake_id "MetaArgAssign", [ E (L lit |> e) ]))
```

Comment:
seems you lost the `name` here, probably just need it as an additional entry in the `any list`